packages/headless/src/features/facets/range-facets/numeric-facet-set/numeric-facet-controller-actions.ts

```diff
@@ -24,14 +26,15 @@ export const executeToggleNumericFacetSelect = createAsyncThunk<
     selection: NumericFacetValue;
   },
   AsyncThunkSearchOptions<ConfigurationSection & NumericFacetSection>
->('numericFacet/executeToggleSelect', ({facetId, selection}, {dispatch}) => {
-  validatePayloadSchema(
-    {facetId, selection},
-    {
-      facetId: facetIdDefinition,
-      selection: new RecordValue({values: numericFacetValueDefinition}),
-    }
-  );
-  dispatch(toggleSelectNumericFacetValue({facetId, selection}));
-  dispatch(executeToggleRangeFacetSelect({facetId, selection}));
+>(executeToggleNumericFacetSelectType, (payload, {dispatch}) => {
```

Comment:
For thunks, an implementation with rejectWithValue is possible, although it clashes with AsyncThunkSearchOptions a bit and will take further refactor. In the end, the result is the same.

```typescript
export const executeToggleNumericFacetSelect = createAsyncThunk<
  void,
  {
    facetId: string;
    selection: NumericFacetValue;
  },
  AsyncThunkSearchOptions<ConfigurationSection & NumericFacetSection>
>(
  executeToggleNumericFacetSelectType,
  (payload, {dispatch, rejectWithValue}) => {
    const {error} = validatePayloadSchema(payload, {
      facetId: facetIdDefinition,
      selection: new RecordValue({values: numericFacetValueDefinition}),
    });
    if (error) {
      return rejectWithValue(error);
    }
    dispatch(toggleSelectNumericFacetValue(payload));
    dispatch(executeToggleRangeFacetSelect(payload));
    return;
  }
);
```

Reply:
rejectWithValue will also serialize the error and remove the useful message, although it will be in the action.
Error serialization is not really necessary as we intercept and log those actions in the first middleware.

Reply:
With current implementation
![Screen Shot 2020-12-02 at 8 43 28 AM](https://user-images.githubusercontent.com/4923043/100880382-a68f5d80-347a-11eb-8a6d-2f3328d71dc1.png)

With rejectWithValue
![Screen Shot 2020-12-02 at 8 44 29 AM](https://user-images.githubusercontent.com/4923043/100880436-bad35a80-347a-11eb-91d4-08cb527b1a54.png)

Reply:
I find the first message easier to read compared to the second because there is no need to expand. However, it would be even better if the action name was at the top instead of at the bottom.

On the other hand, the need for an additional `if` statement in either case is not as nice 😞 Could there be a way around it using middleware?

Reply:
UPDATE: I made the validatePayload methods throw in thunks, and that way I can catch it in the middleware 
Good call @samisayegh  ! 💯

Reply:
Another (maybe more elegant solution) would be to have a function to wrap the base redux toolkit `createAsyncThunk` function, which would wrap the inner function that could throw (eg: with `validatePayload`, or any other thing that could throw in the future).

The "wrapper" would take care of try catching/returning the correct `rejectWithValue(error)` with proper format etc. .

So we would remove the "weird-ish" last boolean parameter to control wether to throw or not (we'd actually make it so it always throw).

Do you think it would make sense ?

What I'm proposing is something similar to the `makeAnalyticsAction`, which takes care of the "boilerplate" to build an analytics redux action.

We could call it "makeThrowableAction" or something... 

Thoughts ?

Reply:
I agree the boolean parameter is strange, but I am also wondering, is it needed?